caffe2/CMakeLists.txt

```diff
@@ -1363,6 +1370,12 @@ if(USE_DISTRIBUTED)
       target_compile_definitions(torch_cuda PUBLIC USE_C10D_NCCL)
     endif()
   endif()
+  if(USE_C10D_XCCL)
+    target_compile_definitions(torch_xpu PUBLIC USE_C10D_XCCL)
+    set_source_files_properties(
+      ${TORCH_SRC_DIR}/csrc/distributed/c10d/ProcessGroupXCCL.cpp
+      PROPERTIES COMPILE_DEFINITIONS  "CCL_ENABLE_ZE;CCL_ENABLE_SYCL")
```

Comment:
for my education, what do "CCL_ENABLE_ZE;CCL_ENABLE_SYCL" stand for?

Reply:
Those flags are just workaround and will be removed after oneCCL gcc build issue is fixed. Without these two flags, C++ cannot compile oneCCL. They will be removed later.

Reply:
@Chao1Han , please add some comments to elaborate on the motivation.

Reply:
Sure.

Reply:
Those flag is needed in oneCCL to find correct header. We will define those flags in XCCL backend file instead of passing to `gcc` compiler.